src/Bootstrappers/PrefixCacheTenancyBootstrapper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Cache\Repository;
+use Illuminate\Support\Facades\Cache;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class PrefixCacheTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected null|string $originalPrefix = null;
+    protected string $storeName;
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        $this->originalPrefix = config('cache.prefix');
+        $this->storeName = config('cache.default');
```

Comment:
Use proper DI in classes like this

Reply:
@abrardev99 By proper DI I didn't mean inject `Application` and then use `$this->app['config']`: that has no benefits of DI — it doesn't list the dependencies, and it doesn't provide any meaningful IDE support or information that phpstan could use.

I mean actually injecting *the dependencies that we use*, not just `Application`.

Reply:
Makes sense. I DI the Config class. And I think that's all the dependencies we are depending on.